cli/pkg/workspace/init.go

```diff
@@ -0,0 +1,111 @@
+package workspace
+
+import (
+	"fmt"
+	"github.com/spf13/viper"
+	"io/fs"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+func Init(path string) ConfigDir {
+	viper.AddConfigPath(getConfigPath(path))
+	viper.SetConfigType("yaml")
+	viper.SetConfigName(cliConfigFileName)
+
+	if err := viper.ReadInConfig(); err != nil {
+		if _, ok := err.(viper.ConfigFileNotFoundError); ok {
+			fmt.Println(err)
+			fmt.Println("the current directory is not an airy config directory")
+		} else {
+			fmt.Println("invalid configuration: ", err)
+		}
+
+		os.Exit(1)
+	}
+
+	dir := ConfigDir{Path: path}
+
+	if _, err := os.Stat(dir.GetAiryYaml()); os.IsNotExist(err) {
+		fmt.Println("the current directory is not an airy config directory")
+		os.Exit(1)
+	}
+	return dir
+}
+
+func getConfigPath(path string) string {
+	if path == "" {
+		path = "."
+	} else if filepath.IsAbs(path) {
+		return path
+	}
+
+	path, err := filepath.Abs(path)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+	return path
+}
+
+
+func Create(path string) (ConfigDir, error) {
+	path = getConfigPath(path)
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		err = os.MkdirAll(path, 0755)
+		if err != nil {
+			return ConfigDir{}, err
+		}
+	}
+
+	entries, err := templateDir.ReadDir("template")
+	if err != nil {
+		return ConfigDir{}, err
+	}
+
+	for _, entry := range entries {
+		if err := recCopy(path, "template", entry); err != nil {
+			return ConfigDir{}, err
+		}
+	}
+
+	// Init viper config
+	err = viper.WriteConfigAs(filepath.Join(path, cliConfigFileName))
```

Comment:
```
	if err != nil {
		return ConfigDir{}, err
	}
```

and then 

```
return ConfigDir{Path: path}, nil
```

would be more precise no? (as in you deal with the error right away)

Reply:
I will move the three commands in between above this one so the immediate return makes more sense :)